tests/framework/utils_imdsv2.py

```diff
@@ -0,0 +1,72 @@
+"""
+A simple IMDSv2 client
+
+- https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instancedata-data-retrieval.html
+- https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instancedata-data-categories.html
+"""
+
+import sys
+import time
+
+import requests
+
+class IMDSv2Client:
+    IMDSV2_HEADER_TOKEN_TTL = "X-aws-ec2-metadata-token-ttl-seconds"
+    IMDSV2_HEADER_TOKEN = "X-aws-ec2-metadata-token"
+
+    def __init__(self, endpoint="http://169.254.169.254/latest"):
+        self.endpoint = endpoint
+        self.ttl = 21600
+        self.token_expiry_time = 0
+        self.token = None
+
+    def get_token(self):
+        """Get a token from IMDSv2"""
+        if self.token_expiry_time < time.time():
+            headers = {self.IMDSV2_HEADER_TOKEN_TTL: str(self.ttl)}
+            url = f"{self.endpoint}/api/token"
+            print(url, headers, file=sys.stderr)
+            res = requests.put(url, headers=headers)
+            self.token = res.content
+            self.token_expiry_time = time.time() + self.ttl
+        return self.token
+
+    def get(self, path):
+        """
+        Get a metadata path from IMDSv2
+
+        >>> IMDSv2Client().get("/meta-data/instance-type")
+        """
+        headers = {self.IMDSV2_HEADER_TOKEN: self.get_token()}
+        url = self.endpoint + path
+        res = requests.get(url, headers=headers)
+        if res.status_code != 200:
+            raise Exception(f"IMDSv2 returned {res.status_code} for {url}")
+        return res.text
+
+
+IMDS_V2 = IMDSv2Client()
+
+
+import subprocess
+
+IMDSV2_HDR_TOKEN_TTL = "X-aws-ec2-metadata-token-ttl-seconds"
```

Comment:
Is it possible to have these constants defined once globally? (So we can drop the duplicated definition in the `IMDSv2Client` as well)

Reply:
Yes, I was trying something that wasn't intended to show in this PR. I will clean up before taking the PR out of draft.